features/check-plugin-update.feature

```diff
@@ -25,3 +25,19 @@ Feature: Check whether plugins are up to date
     Then STDOUT should be a table containing rows:
       | name          | status  | message                                 |
       | plugin-update | warning | 1 plugin has an update available.       |
+
+  Scenario: One plugin has an update available and the failure_status is 'error'
+    Given a WP install
+    And a config.yml file:
+      """
+      plugin-update:
+        check: Plugin_Update
+        options:
+          status_for_failure: error
```

Comment:
I think my preference would be to make this a binary option: `use_error_for_failure`. The default value would be `false`, and it would only exist on the checks that have 'warning' as the failure status. Also, 'failure' isn't quite the best word here. Can you think of others?

Reply:
How about just adding a `--strict` parameter to the command, and have all warning turn into errors that way? Not as granular, but much simpler to implement and comprehend.

Reply:
I agree "failure" isn't the best, I'll see if anything else comes to mind. One reason I left it non--binary was that I was thinking that there may be some checks that use "error" by default and the user would rather them be "warning". For example if someone is using the `Option_Value` check to implement something of their own, that would give them an easy way to create their own warnings ("warning if siteicon isn't set"). Thoughts on that?

@schlessera at least for my use case, `--strict` wouldn't be granular enough to be useful. I could see how it could be a useful option to have in general, but for me I'm looking for per-install and per-check customization (I don't want *all* current warnings to be errors, and want to simply be able to use `wp doctor check --all --config=config.yml` without writing custom commands for each site).

Reply:
Maybe a related question I have is: how would you prefer people extend doctor-command? I can't find the docs at the moment but you can also write your own classes, right? As you can probably tell, I'm more interested in the idea of giving a little more power in the config file. I feel like there's a lot of useful checks you can create with some of the basic classes you already have and maybe a couple config options (warning vs error, and on `Option_Value` I have a `value_is_not` option to test inequality instead). As a user I'm way more interested in tweaking a YAML file then I am writing my own PHP classes, tests, repo, etc.

Reply:
I'm not particularly a fan of `--strict`. I don't think it's quite the right implementation for this need.

> As you can probably tell, I'm more interested in the idea of giving a little more power in the config file.

Yes, the original intent is that Doctor is 80% configurable through the config file, and then supports the additional 20% configuration through class extension when you ned it. Obviously, the interpretation of those percentages is subjective.

> I can't find the docs at the moment but you can also write your own classes, right? 

Here's a good starting point for the documentation: https://danielbachhuber.com/tip/customize-doctor-config/